### PR TITLE
fix: get exact version of `stable`

### PR DESCRIPTION
**Pull request by vil02**
This PR should solve the problem mentioned [here](https://github.com/jiro4989/setup-nim-action/issues/483#issuecomment-2213613289).

### CODE REVIEW

File: install_nim.sh [hunk 1]
```diff
@@ -57,6 +57,11 @@ while ((0 < $#)); do
   esac
 done
 
+# get exact version of stable
+if [[ "$nim_version" = "stable" ]]; then
+    nim_version=$(curl -sSL http://nim-lang.org/channels/stable)
```
Comment (jiro4989) [inline install_nim.sh:43]: Please use `https`, not `http`.

```suggestion
    nim_version=$(curl -sSL https://nim-lang.org/channels/stable)
```

**Review by jiro4989 — CHANGES_REQUESTED** [review]

File: install_nim.sh [hunk 2]
```diff
@@ -74,13 +60,6 @@ if [[ "$nim_version" = "devel" ]]; then
   exit
 fi
 
-# fetch latest tag if 'nim_version' was 'stable'
-if [[ "$nim_version" = "stable" ]]; then
-  # NOTE: jq is pre-installed on github actions runner
-  nim_version="$(fetch_tags | latest_version)"
-elif [[ "$nim_version" =~ ^[0-9]+\.[0-9]+\.x$ ]] || [[ "$nim_version" =~ ^[0-9]+\.x$ ]]; then
```
Comment (jiro4989) [inline install_nim.sh:81]: Please do not delete this block. This block need to process patterns like `1.6.x` and `2.x`. 

```bash
if [[ "$nim_version" =~ ^[0-9]+\.[0-9]+\.x$ ]] || [[ "$nim_version" =~ ^[0-9]+\.x$ ]]; then
  nim_version="$(fetch_tags | grep -E "$(tag_regexp "$nim_version")" | latest_version)"
fi
```

**Review by jiro4989 — CHANGES_REQUESTED** [review]

File: install_nim.sh (hunk 2 above)
Comment (vil02) [reply install_nim.sh:81]: You are right! I was wondering why the CI was failing. Sorry for the confusion.

**Review by jiro4989 — APPROVED** [review]
LGTM 👍 

LGTM

**Pull request merged**